NewSpring/corinth
# feat: Answered Prayer

## DESCRIPTION

### What does this PR do, or why is it needed?
This PR makes it possible for a person to enter an answer to their prayer.

Lots of changes here:
* Changed the `My Prayers` card to show the `ellipsis` icon instead of the trash can. 
* Changed the `My Prayers` card to show the answer to the prayer if there is one.
* Changed the action menu to show "Mark as Answered" if the prayer doesn't have an answer and "Edit Answer" if it does.
* Added a new screen that the person would use to enter in their answer.
* We no longer regard a prayers expiration date on the `My Prayers` (USER) prayer feed. We show all the prayers a person has entered in most recent first order. 
* Prayers with answers, regardless of expiration date should not show in the regular prayer feeds. The exception is the `My Saved Prayers` feed. If you've saved a prayer and it has an answer it will continue to show there. 
* **BONUS**: Fixes the `prayerFeed.edges is not a function` error we would get when deleting a prayer in the sim. Which allows the `My Prayers` feed to correctly update after a prayer has been deleted.

#### Screenshots
The `My Prayers` screen with new icon and answered prayer
![Screenshot 2020-04-30 15 48 42](https://user-images.githubusercontent.com/3229463/80754641-62f7fe00-8afd-11ea-88ee-9969fca29abe.png)

Action menu when tapping the ellipsis icon on a prayer that doesn't have an answer.
![Screenshot 2020-04-30 15 48 49](https://user-images.githubusercontent.com/3229463/80754648-655a5800-8afd-11ea-8c31-28e384b5b887.png)

Action menu when tapping the ellipsis icon on a prayer that has an answer.
![Screenshot 2020-04-30 15 48 56](https://user-images.githubusercontent.com/3229463/80754652-668b8500-8afd-11ea-8733-4f50a9695ee9.png)

New screen to add the answer to your prayer.
![Screenshot 2020-04-30 15 49 15](https://user-images.githubusercontent.com/3229463/80754657-67bcb200-8afd-11ea-9377-deb49743b807.png)

What a prayer with an answer looks like in the `My Saved Prayers` feed.
![Screenshot 2020-04-30 16 06 37](https://user-images.githubusercontent.com/3229463/80754659-67bcb200-8afd-11ea-913b-7b7c0eadad55.png)

### How do I test this PR?
#### Test the Feeds
* Make sure that there are at least 2 prayers for you to pray for (someone else's prayers) set up in `alpha-rock`. One should have an answer and one should not.
* Go to the `My Church` feed and pray for your prayers. You should see the prayer that does not have an answer. You should not see the prayer that has an answer.

#### My Prayers
* Go to `My Prayers`.
* You should see every prayer you have in the system in the list.
* Tap on the ellipsis icon for one of your prayers.
* Choose `Mark as Answered`.
* On the next screen tap the "X" icon to make sure it takes you back to the `My Prayers` list correctly.
* Tap `Mark as Answered` again and this time enter in an answer. The `Save Answer` button should not be enabled until you've entered in some text. This prevents someone from entering a blank answer. 
* Tap `Save Answer`. This should take you back to the `My Prayers` list and the prayer you added an answer for should show that answer now. 
* Tap the ellipsis icon for that same prayer you just entered an answer for. The action menu should now say `Edit Answer`.
* Tap `Edit Answer`. The screen where you entered the answer should show again with the current answer already populated in the input box.
* Change the answer and tap `Save Answer` again. The answer should update and show correctly on the `My Prayers` list page.

#### BONUS: Delete Prayer Fix
* Tap on the ellipsis button on one of the prayers in the `My Prayers` list that you're okay with deleting.
* Choose `Delete`.
* The prayer should delete correctly and the `My Prayers` list should update without an error. 
* NOTE: There might be a flash where it looks like the list is re-rendering. I'm okay with that for now. This error (`prayerFeed.edges is not a function`) is currently happening in `master`, so we can fix any other weirdness we see in a different PR. 

## TODO


## REVIEW


**Manual QA**


---

> The purpose of PR Review is to _improve the quality of the software._
